src/lib/Witch/Instances.hs

```diff
@@ -907,6 +908,16 @@ instance From.From Rational Float where
 instance From.From Rational Double where
   from = fromRational
 
+-- | Uses `fromRational` as long as there isn't a loss of precision.
+instance Fixed.HasResolution a => TryFrom.TryFrom Rational (Fixed.Fixed a) where
+  tryFrom = Utility.eitherTryFrom $ \s ->
+    let
+      t :: Fixed.Fixed a
+      t = fromRational s
+    in if toRational t == s
```

Comment:
I would prefer a more direct way to tell if the conversion would lose precision, but I couldn't think of anything.